### PR TITLE
Updating the js libraries to latest version.

### PR DESCRIPTION
**Pull request by puni9869**
As title.

image

### CODE REVIEW

**Review by delvh — APPROVED** [review]

File: web_src/js/utils.test.js [hunk 1]
```diff
@@ -138,12 +138,12 @@ test('encodeURLEncodedBase64, decodeURLEncodedBase64', () => {
...
+  expect(decodeURLEncodedBase64('QUE/')).deep.equal(uint8array('AA?'));
+  expect(decodeURLEncodedBase64('QUF+')).deep.equal(uint8array('AA~'));
+  expect(decodeURLEncodedBase64('QUE_')).deep.equal(uint8array('AA?'));
+  expect(decodeURLEncodedBase64('QUF-')).deep.equal(uint8array('AA~'));
 
   expect(encodeURLEncodedBase64(uint8array('a'))).toEqual('YQ'); // standard base64: "YQ=="
-  expect(decodeURLEncodedBase64('YQ')).toEqual(uint8array('a'));
-  expect(decodeURLEncodedBase64('YQ==')).toEqual(uint8array('a'));
+  expect(decodeURLEncodedBase64('YQ')).deep.equal(uint8array('a'));
+  expect(decodeURLEncodedBase64('YQ==')).deep.equal(uint8array('a'));
```
Comment (silverwind) [inline web_src/js/utils.test.js]: Why these changes? As I understand `toEqual` is already performing deep equality checks, unless to checked data is an error, so I prefer to keep using it for everything. From [Reference](https://vitest.dev/api/expect.html#toequal):

> A deep equality will not be performed for Error objects. To test if something was thrown, use [toThrowError](https://vitest.dev/api/expect.html#tothrowerror) assertion.

Comment (delvh) [reply web_src/js/utils.test.js]: My guess: Because we are using an `array` here, and not a string like above.
Hence, only the array pointer would be compared and not the array contents.

Comment (wxiaoguang) [reply web_src/js/utils.test.js]: > Hence, only the array pointer would be compared and not the array contents.

Really? Difference arrays usually mean different "pointer", how did the tests pass?

Comment (delvh) [reply web_src/js/utils.test.js]: :shrug: as I said, that's only my guess why these changes were necessary.
@puni9869 will know why they were needed. Perhaps jest made a breaking change in this minor update?

Comment (silverwind) [reply web_src/js/utils.test.js]: > > Hence, only the array pointer would be compared and not the array contents.
> 
> Really? Difference arrays usually mean different "pointer", how did the tests pass?

`toEqual` compares structure only via essential "deep equality". To compare reference, there is [`toBe`](https://vitest.dev/api/expect.html#tobe). In any case, I think the change is not needed.

Comment (puni9869) [reply web_src/js/utils.test.js]: `.toBe` is failed

and Unit test cases fixes: https://github.com/jestjs/jest/issues/9381

I edited the PR description before.

Comment (puni9869) [reply web_src/js/utils.test.js]: After updating the libs and running `make test`
we are getting assertion failed 
image

And as @delvh  mentioned its array so needs deep equaliency check.

Comment (silverwind) [reply web_src/js/utils.test.js]: This does seem like a bug in Vitest's matcher or whatever is providing the UInt8Array implementation, need to investigate. Those UInt8Array are structurally equal so `toEqual` should work.

Comment (puni9869) [reply web_src/js/utils.test.js]: Sure, I will share my findings, No worries.

Comment (wxiaoguang) [reply web_src/js/utils.test.js]: Yes, `toEqual` is the right function, the old code is right, it is an upstream bug.

->  0.34.3 doens't work with Uint8Array and TextEncoder https://github.com/vitest-dev/vitest/issues/4043

File: package.json [hunk 2]
```diff
@@ -79,9 +79,9 @@
     "stylelint-declaration-strict-value": "1.9.2",
     "stylelint-stylistic": "0.4.3",
     "svgo": "3.0.2",
-    "updates": "14.3.5",
+    "updates": "14.4.0",
     "vite-string-plugin": "1.1.2",
-    "vitest": "0.34.2"
+    "vitest": "0.34.3"
```
Comment (wxiaoguang) [inline package.json]: It could keep vitest back to 0.34.2, until upstream resolves their bug

Comment (silverwind) [reply package.json]: Yes, please run `npm i -D vitest@0.34.2` on this branch and then revert the test changes.

**Review by silverwind — APPROVED** [review]
